codegen/src/test/kotlin/software/amazon/smithy/rust/lang/RustWriterTest.kt

```diff
@@ -16,81 +16,94 @@ import software.amazon.smithy.model.shapes.SetShape
-import software.amazon.smithy.rust.codegen.testutil.compileAndRun
 import software.amazon.smithy.rust.codegen.testutil.compileAndTest
-import software.amazon.smithy.rust.codegen.testutil.shouldCompile
 import software.amazon.smithy.rust.codegen.testutil.testSymbolProvider
+import software.amazon.smithy.rust.codegen.testutil.unitTest
 import software.amazon.smithy.rust.codegen.util.lookup
 
 class RustWriterTest {
     @Test
     fun `inner modules correctly handle dependencies`() {
-        val sut = RustWriter.forModule("parent")
-        val requestBuilder = RuntimeType.HttpRequestBuilder
-        sut.withModule("inner") {
-            rustBlock("fn build(builder: #T)", requestBuilder) {
+        val project = TestWorkspace.testProject()
+        project.withModule(RustModule.public("parent")) {
+            val requestBuilder = RuntimeType.HttpRequestBuilder
+
+            it.withModule("inner") {
+                rustBlock("fn build(builder: #T)", requestBuilder) {}
             }
+            val httpDep = CargoDependency.Http.dependencies[0]
+            it.dependencies shouldContain httpDep
+            it.toString() shouldContainOnlyOnce "DO NOT EDIT"
         }
-        val httpDep = CargoDependency.Http.dependencies[0]
-        sut.dependencies shouldContain httpDep
-        sut.toString() shouldContainOnlyOnce "DO NOT EDIT"
+        project.compileAndTest()
     }
 
     @Test
     fun `manually created struct`() {
-        val sut = RustWriter.forModule("lib")
-        val stringShape = StringShape.builder().id("test#Hello").build()
-        val set = SetShape.builder()
-            .id("foo.bar#Records")
-            .member(stringShape.id)
-            .build()
-        val model = Model.assembler()
-            .addShapes(set, stringShape)
-            .assemble()
-            .unwrap()
+        val project = TestWorkspace.testProject()
+        project.lib {
+            val stringShape = StringShape.builder().id("test#Hello").build()
+            val set = SetShape.builder()
+                .id("foo.bar#Records")
+                .member(stringShape.id)
+                .build()
+            val model = Model.assembler()
+                .addShapes(set, stringShape)
+                .assemble()
+                .unwrap()
 
-        val provider: SymbolProvider = testSymbolProvider(model)
-        val setSymbol = provider.toSymbol(set)
-        val stringSymbol = provider.toSymbol(stringShape)
-        sut.rustBlock("struct Test") {
-            write("member: #T,", setSymbol)
-            write("otherMember: #T,", stringSymbol)
+            val provider: SymbolProvider = testSymbolProvider(model)
+            val setSymbol = provider.toSymbol(set)
+            val stringSymbol = provider.toSymbol(stringShape)
+            it.rustBlock("struct Test") {
+                write("member: #T,", setSymbol)
+                write("otherMember: #T,", stringSymbol)
+            }
+            it.rustBlock("fn inner()") {
```

Comment:
We cannot do this within `unitTest` because it takes a `string` rather than provide a writer.

Reply:
https://github.com/awslabs/smithy-rs/pull/1450 addresses this

Reply:
Fixed https://github.com/awslabs/smithy-rs/pull/1446/commits/b8c70b83bb0dd26554303b99888768109a68d44d